modules/centrifuge/main.nf

```diff
@@ -0,0 +1,57 @@
+process CENTRIFUGE {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::centrifuge=1.0.4_beta" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/centrifuge:1.0.4_beta--h9a82719_6' :
+        'quay.io/biocontainers/centrifuge:1.0.4_beta--h9a82719_6' }"
+
+    input:
+    tuple val(meta), path(reads)
+    path db
+    val save_unaligned
+    val sam_format
+
+    output:
+    tuple val(meta), path('*report.txt')  , emit: report
+    tuple val(meta), path('*results.txt') , emit: results
+    tuple val(meta), path('*kreport.txt') , emit: kreport
+    tuple val(meta), path('*.sam')        , optional: true, emit: sam
+    tuple val(meta), path('*.fastq.gz')   , optional:true, emit: fastq
+    path "versions.yml"                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def paired = meta.single_end ? "-U ${reads}" :  "-1 ${reads[0]} -2 ${reads[1]}"
+    def db_name = db.toString().replace(".tar.gz","")
+    def unaligned = ''
+    if (meta.single_end) {
+        unaligned = save_unaligned ? "--un-gz ${prefix}.unmapped.fastq.gz" : ''
+    } else {
+        unaligned = save_unaligned ? "--un-conc-gz ${prefix}.unmapped.fastq.gz" : ''
+    }
+    def sam_output = sam_format ? "--out-fmt 'sam'" : ''
+    """
+    tar -xf $db
+    centrifuge \\
+        -x $db_name \\
+        -p $task.cpus \\
+        $paired \\
+        --report-file ${prefix}.centrifuge.report.txt \\
+        -S ${prefix}.centrifuge.results.txt \\
```

Comment:
You shouldn't specify anything the output file name other than `${prefix}.<extension>`, as this depends on the pipeline user.

In this case as you have multiple `txt` files, you could instead go `${prefix}.report` and `${prefix}.results` etc. (make sure to update the output block).

Last point here: https://nf-co.re/developers/adding_modules#naming-conventions

Reply:
Thanks for the further clarification.